[JWY-220303] member

Wooyongjeong opened the pull request:
- MemberService 및 테스트 코드 작성
- CreateMemberForm, UpdateMemberForm 추가
- Member 엔티티에 연관관계메서드 작성
- Member 엔티티에 @Embedded를 이용하여 PhoneNumber 추가 및 기존 TestDataInit과 테스트 코드에서 PhoneNumber 넣어주는 방식 변경

Wooyongjeong committed:
[JWY-220303]
- MemberService 및 테스트 코드 작성
- CreateMemberForm, UpdateMemberForm 추가
- Member 엔티티에 연관관계메서드 작성
- Member 엔티티에 @Embedded를 이용하여 PhoneNumber 추가 및 기존 TestDataInit과 테스트 코드에서 PhoneNumber 넣어주는 방식 변경

LimHyungang commented on src/main/java/com/ootd/with/domain/member/Member.java:
```diff
@@ -72,31 +67,51 @@
+    }
+
+    //Member 탈퇴
+    public void delete() {
```
changeStatus(StatusType type) 으로 형식 맞추기로 했었는데 반영이 안 됐네요

Wooyongjeong replied:
형식에 맞춰 수정하였습니다.

LimHyungang commented on src/main/java/com/ootd/with/domain/member/Member.java:
```diff
@@ -72,31 +67,51 @@
+    }
+
+    //Member 탈퇴
+    public void delete() {
```
changeStatus(StatusType type) 으로 형식 맞추기로 했었는데 반영이 안 됐네요

LimHyungang replied:
changeStatus(StatusType type) 으로 형식 맞추기로 했었는데 반영이 안 됐네요

LimHyungang commented on src/main/java/com/ootd/with/domain/member/MemberServiceImpl.java:
```diff
@@ -13,10 +15,41 @@
+    }
+
+    @Override
+    public void update(UpdateMemberForm form) {
```
이 부분 통일이 필요합니다.

저는 UpdateForm에 수정 화면에서 유저가 입력하는 정보만을 담았습니다.
id는 유저에게는 드러나지 않는 값이기 때문에 UpdateForm에 포함시키지 않았어요. 

컨트롤러에서 path variable로 id를 받았고 이걸 통해 엔티티를 조회하여 service.update(entity, form)을 호출하도록 했습니다. (BoardController.updateBoard() 참고)

만약 지금같은 방식을 고수한다면 UpdateForm에 id값은 어떻게 전달해줄 생각인가요?

Wooyongjeong replied:
네 id값을 따로 파라미터로 받는 것으로 수정하였습니다.

LimHyungang commented on src/main/java/com/ootd/with/domain/member/PhoneNumber.java:
```diff
@@ -0,0 +1,29 @@
+@NoArgsConstructor
+@Getter
+@Embeddable
+public class PhoneNumber {
```
각 번호 덩어리에 대한 자리수 검증 로직을 추가하는게 좋지 않을까요?

ex)
firstPhoneNumber : 3자리
midPhoneNumber : 3 or 4 자리
lastPhoneNumber : 4자리

클라이언트에서 입력 방식에 제한을 두겠지만 이런 비슷한 경우가 발생했을때 서버에서 따로 검증을 거치기로 얘기했던 것 같아서요.

Wooyongjeong replied:
javax.validation.constraints 패키지의 `@Size` 애노테이션을 이용하여 반영하였습니다.

Wooyongjeong committed:
[JWY-220305]
- Member의 연관관계 메소드인 delete() -> changeStatus(StatusType statusType)으로 변경
- MemberService의 update() 함수의 파라미터인 UpdateMemberForm에 id를 포함하여 전달하는 것에서 Long memberId 파라미터를 따로 받는 것으로 변경
- PhoneNumber에 @Size validation 추가

Wooyongjeong committed:
[JWY-220306]
- MemberService의 update() 함수에서 파라미터로 Long memberId를 받지 않고 Member member로 받는 것으로 변경
